packages/approval-controller/src/ApprovalController.ts

```diff
@@ -13,13 +13,24 @@ const controllerName = 'ApprovalController';
 type ApprovalPromiseResolve = (value?: unknown) => void;
 type ApprovalPromiseReject = (error?: unknown) => void;
 
-type ApprovalRequestData = Record<string, Json> | null;
+export type ApprovalRequestData = Record<string, Json> | null;
 
 type ApprovalRequestState = Record<string, Json> | null;
 
 type ApprovalCallbacks = {
   resolve: ApprovalPromiseResolve;
   reject: ApprovalPromiseReject;
+  sideEffects?: SideEffects;
+};
+
+export type SideEffectParams = {
+  requestData: Record<string, Json> | null;
+  messagingSystem: SideEffectMessenger;
+};
+
+export type SideEffects = {
+  permittedHandlers: ((params: SideEffectParams) => Promise<unknown>)[];
```

Comment:
Is it overkill to have a `SideEffect` type for `(params: SideEffectParams) => Promise<unknown>` to avoid this duplication?

Reply:
Indeed it could be nice